nfqueue_gteq_1.12.go

```diff
@@ -117,23 +120,27 @@ func (nfqueue *Nfqueue) setVerdict(id uint32, verdict int, batch bool, attribute
 }
 
 func (nfqueue *Nfqueue) socketCallback(ctx context.Context, fn HookFunc, errfn ErrorFunc, seq uint32) {
+	defer nfqueue.wg.Done()
 	defer func() {
 		// unbinding from queue
 		_, err := nfqueue.setConfig(uint8(unix.AF_UNSPEC), seq, nfqueue.queue, []netlink.Attribute{
 			{Type: nfQaCfgCmd, Data: []byte{nfUlnlCfgCmdUnbind, 0x0, 0x0, byte(nfqueue.family)}},
 		})
 		if err != nil {
 			nfqueue.logger.Printf("Could not unbind from queue: %v\n", err)
-			return
 		}
 	}()
+
+	nfqueue.wg.Add(1)
```

Comment:
As you have described, there can be situations where this Waiting group is needed.
But I'm not sure about the additional one added here before the call to `nfqueue.socketCallback()`.
https://github.com/capnspacehook/go-nfqueue/blob/a6bd93375f2199107630cf6e211bc0b19f8dde09/nfqueue.go#L148-L149

This feels like a double net for safty that adds complexity.

Reply:
Honestly that's what seems the most important to me. I created this PR because I wanted a way to ensure that when `nfqueue.Close()` returns, no instances of my callback function I passed in to `RegisterWithErrorFunc` will be running.

Reply:
Looking at the code again I think the complexity got reduced with the removal of the Go 1.12 legacy. Now `Add()` and `Done()` are no longer distributed across different files.
So it looks like it should be fine.

Can you update the PR and rebase it on top of the latest changes?

Reply:
Ah, I didn't totally understand what you found complex but that makes sense now. Sure I'll do that soon!